### ISSUE TITLE
Dependency error when moving Data Type out of a folder

### ISSUE THREAD

**Issue by HamDerAndrew** [umbraco/Umbraco.Deploy.Issues#125]
If you move a data type out of a folder and then delete that folder, you will hit a dependency error, when content transfering from the source environment to target environment

Reproduction
------------

### Bug summary
If you create a data type and move it into a folder, and then deploy this to the target environment, you will hit a dependency error if you later decide to move the data type out of the folder and then deleting that folder.

### Specifics
Happens for Deploy 4.5.0.0.

### Steps to reproduce
This is tested between 2 Cloud environments, but a customer experienced it in a Local to Dev scenario.

1. Go to the Backoffice on your Dev environment
2. Create a "folder" under Data Types. Name it what you want
3. Create a Data Type in this folder
4. Use it on a Document Type
5. Create a content node based on the Document Type, which has the Data Type. Save and Publish it
6. Deploy code changes to target environment (Live in this case)
7. Content Transfer the content you made.
8. Now, on the Dev environment, move the Data Type out of the folder
9. Deploy the code change that this causes, from Dev to the target environment
10. Go to the Content section on Dev and modify the content on the document type that has the data type. Save and Publish it.
11. Content Transfer it. You now hit a schemamismatch (I guess it makes sense because we did change schema afterall)
12. On Dev, navigate to "Settings" and delete the folder. Attempt to Content Transfer again. You should now hit a dependency error for the folder.

To fix it, you can create a folder on Dev again and name it whatever you like.
Then move the Data Type in there.
Resave the Data Type and deploy these code changes from Dev to Live.
You can now content transfer successfully.

### Expected result

You should be able to content transfer without hitting a dependency error on a folder, given that you deployed these code changes prior to content transfering.

### Actual result
This is an example of what happens if I move the data type out of a folder, delete the folder, deploy the code change and then try to content transfer content that uses the data type:

Dependency Error
```
The operation cannot be completed due to dependencies that cannot be found for 1 item selected for transfer.

There are dependencies on 1 item to resolve.

Dependency Details
Item selected for transfer: Cool Data Type
Id: 1075
Udi: umb://data-type/4ed4d05e5f6d44d9bfcded0802a3e75c
Type: data-type
Depends on:
Udi: umb://data-type-container/69176de8db6c4f4b83db1e6291e6dd8c
Type: data-type-container
Dependency does not exist.
These issues must be resolved before the transfer operation can succeed. For more information, please refer to this documentation page.
```

**Comment by dzordzs** [comment]
I can reproduce this on 4.6 as well.

**Comment by AndyButland** [comment]
Thanks for the write-up, very clear.  We'll take a look (though just to note, I am away for a week).  Meantime I think you could also resolve this is you saved the datatype after moving it.  My expectation is that saving the data type would remove the dependency on the folder if it's not in one.  But I guess the "move" action isn't enough to trigger this (but should be).

**Comment by AndyButland** [comment]
I've replicated these steps.  I find if I run a "Clear Cached Signatures" operation on the dev environment after step 11, the transfer goes through OK.  This suggests the "move" operation isn't triggering an update of the signatures as it should, so whilst the one on the live site is updated, the one on the dev site isn't.

I've replicated this locally now too and have a fix for the next patch update.
